src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -20,6 +21,20 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer
 {
     internal class DefaultRazorDocumentMappingService : RazorDocumentMappingService
     {
+        private readonly ILogger? _logger;
+
+        public DefaultRazorDocumentMappingService(ILoggerFactory loggerFactory) : base()
```

Comment:
Why not just make this parameter null? If everyone has to null check the `_logger`, does having a separate constructor with a non-null parameter add any value?

Reply:
Moq complained that `DefaultRazorDocumentMappingService` needs to have a parameter-less constructor. Maybe calling the empty constructor was a bit sassy on my-part but I generally like to have all constructors "chain" when I can, just in case it matters.

Reply:
I didn't even notice the `base` call, it was just odd seeing a non-null parameter being assigned to a null field.. means every field usage still needs to be checked etc. so the non-null is not doing much.

Reply:
You're right, the correct fix here is to disable CS8618 for the parameter-less constructor and mark it obsolete.

Reply:
The Moq behaviour is surprising, but I generally don't like mocks anyway, so happy to add another reason to the list :)